api/karpenter/v1beta1/karpenter_types.go

```diff
@@ -0,0 +1,414 @@
+	// +optional
+	AssociatePublicIPAddress *bool `json:"associatePublicIPAddress,omitempty"`
+
+	// Tags to be applied on ec2 resources like instances and launch templates.
+	// +kubebuilder:validation:XValidation:message="empty tag keys aren't supported",rule="self.all(k, k != '')"
+	// +kubebuilder:validation:XValidation:message="tag contains a restricted tag matching eks:eks-cluster-name",rule="self.all(k, k !='eks:eks-cluster-name')"
+	// +kubebuilder:validation:XValidation:message="tag contains a restricted tag matching kubernetes.io/cluster/",rule="self.all(k, !k.startsWith('kubernetes.io/cluster') )"
+	// +kubebuilder:validation:XValidation:message="tag contains a restricted tag matching karpenter.sh/nodepool",rule="self.all(k, k != 'karpenter.sh/nodepool')"
+	// +kubebuilder:validation:XValidation:message="tag contains a restricted tag matching karpenter.sh/nodeclaim",rule="self.all(k, k !='karpenter.sh/nodeclaim')"
+	// +kubebuilder:validation:XValidation:message="tag contains a restricted tag matching karpenter.k8s.aws/ec2nodeclass",rule="self.all(k, k !='karpenter.k8s.aws/ec2nodeclass')"
+	// +optional
+	Tags map[string]string `json:"tags,omitempty"`
+
+	// BlockDeviceMappings to be applied to provisioned nodes.
+	// +kubebuilder:validation:XValidation:message="must have only one blockDeviceMappings with rootVolume",rule="self.filter(x, has(x.rootVolume)?x.rootVolume==true:false).size() <= 1"
+	// +kubebuilder:validation:MaxItems:=50
+	// +optional
+	BlockDeviceMappings []*BlockDeviceMapping `json:"blockDeviceMappings,omitempty"`
+
+	// InstanceStorePolicy specifies how to handle instance-store disks.
+	// +optional
+	InstanceStorePolicy *InstanceStorePolicy `json:"instanceStorePolicy,omitempty"`
+
+	// DetailedMonitoring controls if detailed monitoring is enabled for instances that are launched
+	// +optional
+	DetailedMonitoring *bool `json:"detailedMonitoring,omitempty"`
+
+	// MetadataOptions for the generated launch template of provisioned nodes.
+	//
+	// This specifies the exposure of the Instance Metadata Service to
+	// provisioned EC2 nodes. For more information,
+	// see Instance Metadata and User Data
+	// (https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html)
+	// in the Amazon Elastic Compute Cloud User Guide.
+	//
+	// Refer to recommended, security best practices
+	// (https://aws.github.io/aws-eks-best-practices/security/docs/iam/#restrict-access-to-the-instance-profile-assigned-to-the-worker-node)
+	// for limiting exposure of Instance Metadata and User Data to pods.
+	// If omitted, defaults to httpEndpoint enabled, with httpProtocolIPv6
+	// disabled, with httpPutResponseLimit of 1, and with httpTokens
+	// required.
+	// +kubebuilder:default={"httpEndpoint":"enabled","httpProtocolIPv6":"disabled","httpPutResponseHopLimit":1,"httpTokens":"required"}
+	// +optional
+	MetadataOptions *MetadataOptions `json:"metadataOptions,omitempty"`
```

Comment:
I don't we want to expose this, instead we can just mirror to the default required

Reply:
why not, we expose the option in the NodePool API (via annotation)